craft_parts/plugins/scons_plugin.py

```diff
@@ -0,0 +1,131 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright 2022 Canonical Ltd.
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License version 3 as published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""The SCons plugin."""
+
+from typing import Any, Dict, List, Optional, Set, cast
+
+from .. import errors
+from . import validator
+from .base import Plugin, PluginModel, extract_plugin_properties
+from .properties import PluginProperties
+
+
+class SConsPluginProperties(PluginProperties, PluginModel):
+    """The part properties used by the SCons plugin."""
+
+    scons_options: List[str] = []
```

Comment:
Other plugins seem to use `<name>_parameters`, we may want to follow this pattern for consistency.

Reply:
yes, I would make things general change if possible

Reply:
I originally used `scons_options` to mirror the original plugin in snapcraft, but I'm fine with `scons_parameters`. I'll update it

Reply:
updated